### PR TITLE
feat(profiling)fetch function metrics when expanding the function in the slowest function list

### PR DESCRIPTION
**Pull request by JonasBa**
Add the data fetching hook and expand/close functionality

### CODE REVIEW

File: static/app/utils/profiling/hooks/useProfilingFunctionMetrics.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const path = `/organizations/${organization.slug}/events-stats/`;
+  const endpointOptions = useMemo(() => {
+    const params: {
+      query: Record<string, any>;
+    } = {
+      query: {
+        project: props.projects,
+        environment: selection.environments ?? selection.environments,
+        ...normalizeDateTimeParams(selection.datetime ?? selection.datetime),
+        dataSource: 'profilesMetrics',
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useProfilingFunctionMetrics.tsx:27]: I believe this is called `profileFunctionsMetrics`

File: static/app/utils/profiling/hooks/useProfilingFunctionMetrics.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        fingerprint: props.fingerprint,
+        query: props.query,
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useProfilingFunctionMetrics.tsx:29]: I dont think it takes a fingerprint as a query param. Instead, you should be passing the fingerprint inside the query like
```suggestion
        query: `${props.fingerprint}`,
```

File: static/app/views/profiling/landing/slowestFunctionsTable.tsx [hunk 3]
```diff
@@ -238,6 +259,29 @@ function SlowestFunctionsProjectBadge(props: SlowestFunctionsProjectBadgeProps)
...
+
+function SlowestFunctionTimeSeries(props: SlowestFunctionTimeSeriesProps) {
+  const projects = useMemo(() => {
+    return props.function.examples
+      .map(f => {
+        if (typeof f !== 'string' && 'project_id' in f) return f.project_id;
+        return null;
+      })
+      .filter(n => n !== null);
+  }, [props.function]);
```
Comment (Zylphrex) [inline static/app/views/profiling/landing/slowestFunctionsTable.tsx:274]: Nit: you may want to dedupe project ids from this list

**Review by Zylphrex — APPROVED** [review]

File: static/app/views/profiling/landing/slowestFunctionsTable.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/views/profiling/landing/slowestFunctionsTable.tsx:274]: Ah yes, right. Let me do that

**Pull request merged**